ledger-api/grpc-definitions/com/daml/ledger/api/v2/state_service.proto

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "com/daml/ledger/api/v1/ledger_offset.proto";
+import "com/daml/ledger/api/v2/reassignment.proto";
+import "com/daml/ledger/api/v2/transaction_filter.proto";
+
+option java_outer_classname = "StateServiceOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// Allows clients to get state from the ledger.
+service StateService {
+
+  // Returns a stream of the snapshot of the active contracts and incomplete reassignments at a ledger offset.
+  // If there are no active contracts, the stream returns a single response message with the offset at which the snapshot has been taken.
+  // Clients SHOULD use the offset in the last GetActiveContractsResponse message to continue streaming transactions with the update service.
```

Comment:
I like the idea. The mandatory active_at_offset would remove the need of the offset from the message also.
I would suggest maybe to do that only for 3.0 because it changes slightly the logic of this endpoint.
@rgugliel-da @cocreature WDYT?

Reply:
Making the valid at required is a breaking change so if we want to go that road, we should do it know. 
I kind of like the idea of having it optional because it makes the endpoint easier to use and I think that quite a few clients will not specify an offset for the bootstrapping. But no strong opinion.

Reply:
happy to have it be mandatory. We already always use the option with specifying an offset.